java/core/src/main/java/com/microsoft/bond/BondType.java

```diff
@@ -245,6 +245,9 @@ protected final void serializeSomethingField(
             StructBondType.StructField<T> field) throws IOException {
         if (value != null) {
             serializeField(context, value.getValue(), field);
+        } else if (!field.isOptional()) {
```

Comment:
For my understanding, is this the method that will enforce this behavior for strings and collections?